featurebyte/service/base_document.py

```diff
@@ -802,7 +818,12 @@ def _get_unique_constraints(
         document_class: Union[Type[Document], Type[DocumentUpdateSchema]],
         original_document: Optional[FeatureByteBaseDocumentModel],
     ) -> Iterator[
-        tuple[dict[str, Any], dict[str, Any], Optional[UniqueConstraintResolutionSignature]]
+        tuple[
+            dict[str, Any],
```

Comment:
can consider returning a dataclass instead of the tuple with 3 dictionaries since that can be hard to read / understand what each dictionary is supposed to represent

Reply:
Will fix it in the coming PR: https://github.com/featurebyte/featurebyte/pull/1893